src/components/task-list/task-list.component.js

```diff
@@ -1,26 +1,42 @@
-import { useState } from "react";
 import styles from "./task-list.module.css";
 
 const TaskList = (props) => {
-  const [completed, setCompleted] = useState([]);
-  const tasks = props.tasks;
+  const { tasks, taskCompletionToggle, handleTaskCompleted } = props;
 
-  const toggleCompletion = (event) => {
-    setCompleted({ ...completed });
+  const completionToggle = (event) => {
+    taskCompletionToggle(event);
+  };
+
+  const clickCheckbox = (clickEvent) => {
+    handleTaskCompleted(clickEvent);
   };
 
   return (
     <ul>
       {tasks.map((task, index) => {
         return (
-          <li className={styles.task} key={index}>
+          <li
+            className={task.completed ? styles.completedTask : styles.task}
+            key={index}
+            value={task.name}
+            onChange={completionToggle}
+            id={task.id}
```

Comment:
as suggested above `id` can potentially be moved to the `input`.

after some refactor (based on my comments above) you'd find that you might not need the `value`, `onChange` or `id` attributes here.

as a rule of thumb, try to avoid as much as possible adding a `onChange` event to elements that are not originally actionable (as it's a bad practice - not too accessible). Only if you really need to, you should also add a role button to the element: `role="button"` (so screen readers can understand that the element has some behaviour attached)

Reply:
5e5a7a29fdb2d187f9c066e9b8190520e4d22d3a Removed those attributes and added some to the input refactoring the code around it.